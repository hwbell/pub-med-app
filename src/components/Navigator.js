import React from 'react';
import '../App.css';
import 'bootstrap/dist/css/bootstrap.css';

// styling
// import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

// routing with react-router
import { Link } from 'react-router-dom';

const links = [
  {
    title: 'About',
    link: '/about'
  },
  {
    title: 'Search',
    link: '/search'
  },
  {
    title: 'Collections',
    link: '/collections'
  },

]

class Navigator extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    // 
  }

  render() {
    return (

      <div className="fixed-top navigator" style={styles.main}>

        <div className="row">
          <a href="#home" id="logo" className="nav-link">PubdddddMed</a>
        </div>

        <div className="row links-holder">
          {links.map((link, i) => {
            return (
              <a className="nav-link" style={styles.link} key={i} href={link.link}>{link.title}</a>
            )
          })}
        </div>


      </div>
    );
  }
}

const styles = {
  main: {
    padding: '10px',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center'
  },
  link: {
    fontSize: '24px'
  }

}

export default Navigator;
