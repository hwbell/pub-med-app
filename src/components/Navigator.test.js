import React from 'react';
import ReactDOM from 'react-dom';
import Navigator from './Navigator';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', async () => {
  let wrapper = mount(<Navigator />);
});

it('renders correctly', async () => {
  const tree = await renderer
    .create(<Navigator/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the correct elements', () => {
  let wrapper = shallow( <Navigator/>);
  expect(wrapper.html()).toContain('div class=\"fixed-top navigator\"');
  expect(wrapper.html()).toContain('a href=\"#home\" id=\"logo\" class=\"nav-link\"')
  expect(wrapper.find('.nav-link').length).toEqual(4);

})