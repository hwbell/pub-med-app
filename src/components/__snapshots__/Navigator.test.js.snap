// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="fixed-top navigator"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "padding": "10px",
    }
  }
>
  <div
    className="row"
  >
    <a
      className="nav-link"
      href="#home"
      id="logo"
    >
      PubMed
    </a>
  </div>
  <div
    className="row links-holder"
  >
    <a
      className="nav-link"
      href="/about"
    >
      About
    </a>
    <a
      className="nav-link"
      href="/search"
    >
      Search
    </a>
    <a
      className="nav-link"
      href="/collections"
    >
      Collections
    </a>
  </div>
</div>
`;
